src/Command/ProcessCommand.php

```diff
@@ -194,10 +233,15 @@ private function getImages(
             ->setParameter('now', $now)
             ->setParameter('maximumNumberOfTries', $this->maximumNumberOfTries)
             ->setParameter('processingStates', [ImageInterface::PROCESSING_STATE_PENDING, ImageInterface::PROCESSING_STATE_FAILED])
-            ->setMaxResults($maxResults)
         ;
 
         do {
+            if ($this->limitPerResource !== null) {
+                $remaining = $this->limitPerResource - $firstResult;
+                $maxResults = min($maxResults, $remaining);
+            }
```

Comment:
Why do we need this?

Why not just:

```php
if ($this->limitPerResource !== null) {
    $qb->setMaxResults($this->limitPerResource);
}
```

Reply:
Becuase if limit is 100_000 we dont wan't to pull that many images from the database in one go, but still do it in bites of `maxResults`

Reply:
Stupid me, sorry. Sometimes code review is difficult when you don't see the full picture, but just laser focus on something that seems like it doesn't make sense